src/components/forms/SubmitTemplate/server-action.ts

```diff
@@ -15,19 +15,9 @@ export async function submitTemplate(formData: FormData) {
   }
 
   try {
-    if (session.templateType === 'UNKNOWN') {
-      throw new Error('Unknown template type');
-    }
+    const templateDTO = createTemplateFromSession(session);
 
-    const templateDTO = templateFromSessionMapper(
-      session.templateType,
-      session
-    );
-
-    const validatedTemplate = validateTemplate(
-      session.templateType,
-      templateDTO
-    );
+    const validatedTemplate = validateTemplate(templateDTO);
```

Comment:
Thinking a little about this, I feel the most pressing place that needs validation is shifting it left into `getSession`. There it's getting the session from the database, but it's getting it's type from an assertion (`return data as Session`), not from any actual validation. 

If the session data you get back from the database isn't valid, you're more liable to see a runtime error when calling `createTemplateFromSession` before you get to this validation. 

And then once you add the validation to the session, it seems like the `validateTemplate` function is kind of redundant, because you've already validated the session, and the template creation logic is entirely within the scope of this app. i.e. Given a valid session, should `createTemplateFromSession` ever return an invalid template?

Reply:
Hm... possibly? I like the fail fast/early and generally agree with you. I think my hesitancy is that the session object can be in different `valid`states depending on the `templateType` you choose. Accounting for each different valid state seems strange? What do you think? maybe another call?

Reply:
Honestly think I'm faffing a bit too much about this. I'm going to leave it as for now and write up some tickets.